packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,580 @@
+
+    if ((newSelectedIndex !== undefined && newSelectedIndex !== null) &&
+      (newSelectedIndex !== this.state.selectedIndex || newProps.swatchColorPickerItems !== this.props.swatchColorPickerItems)) {
+      this.setState({
+        selectedIndex: newSelectedIndex
+      });
+    }
+  }
+
+  public render() {
+    let { swatchColorPickerButtonProps } = this.props;
+
+    // If we got button props, put the swatch color picker behind a button, otherwise
+    // render all of the items
+    let renderElement = swatchColorPickerButtonProps !== undefined ? this._buttonToRender() : this._fullSwatchColorPickerToRender();
+
+    return (
+      <div
+        ref={ this._resolveRef('_buttonWrapper') }
+        className={ css('ms-swatchColorPickerWrapper', styles.wrapper) }>
+        { renderElement }
+      </div>);
+  }
+
+  /**
+   * Get the selected item's index
+   * @param items - The items to search
+   * @param selectedId - The selected item's id to find
+   * @returns {number} - The index of the selected item's id, -1 if there was no match
+   */
+  private _getSelectedIndex(items: ISwatchColorPickerItemProps[], selectedId: string): number {
+    return findIndex(items, (item => (item.id === selectedId)));
+  }
+
+  /**
+   * Render the swatch color picker with a button in the given (collapased vs expanded state)
+   * @returns {JSX.Element[]} - an array of the elements to render
+   */
+  private _buttonToRender(): JSX.Element[] {
+    let { selectedIndex, expanded } = this.state;
+    let {
+      swatchColorPickerButtonProps,
+      swatchColorPickerItems,
+      updateButtonIconWithColor,
+      width,
+      disabled
+    } = this.props;
+
+    let colorToSet = null;
+
+    // Do we need to update the button with the selected
+    // item's color? If so, attempt to grab the color
+    if (updateButtonIconWithColor &&
+      selectedIndex &&
+      selectedIndex >= 0 &&
+      selectedIndex < swatchColorPickerItems.length) {
+      let swatchColorPickerItem = swatchColorPickerItems[selectedIndex];
+      if (swatchColorPickerItem.type === SwatchColorPickerItemType.Cell &&
+        swatchColorPickerItem.color !== null &&
+        swatchColorPickerItem.color.length > 0) {
+        colorToSet = swatchColorPickerItem.color;
+      }
+    }
+
+    let elements: JSX.Element[] = [];
+
+    elements.push(<DefaultButton
+      { ...swatchColorPickerButtonProps }
+      style={ { color: colorToSet && colorToSet } }
+      className={
+        css('ms-swatchColorPickerButton',
+          {
+            'is-expanded': expanded
+          }
+        )
+      }
+      onClick={ this._onButtonClick }
+      aria-haspopup={ true }
+      aria-expanded={ (!this.props.disabled && this.state.expanded) ? true : false }
+      disabled={ this.props.disabled }
+      menuIconProps={ !swatchColorPickerButtonProps.menuIconProps ? { iconName: 'chevronDown' } : swatchColorPickerButtonProps.menuIconProps }
```

Comment:
I think the default nomenclature here would be to not do the negated check, but instead check swatchColorPickerButtonProps.menuIconProps